notebook/06_xgboost_trade.ipynb

```diff
@@ -0,0 +1,841 @@
olBTU4MePXrA09MTY8aMwVNPPQUzMzO5o7YrH3zwAT777DNkZGTIHeWhZGRk4NChQzh58iRCQ0ORnZ0Nc3NzjBw5EmPGjMGoUaPQu3dvjhKldq+yshIXLlxASEgIQkJCcP78eVRWVsLFxQWjRo3CuHHjMGHCBE7NJZMLFy5g3rx5yMjIwMaNG7FgwQJOXUhERERE7REL7e1RamoqTp8+jTNnzuCPP/5AWloatLW10b17d7i7u0sbi+/UHpWWliIqKgqRkZHSFh8fj9raWnTp0gXjxo3DsGHDMHr0aDg4OMgdt12bNGkS9PT08PPPP8sdpVGkpqbi+PHjOH78OE6ePInc3FwYGxujb9++am2zm5ub3FGJmpTq7yl12+GKigrY2dlh+PDhGDduHJ588slW802W9qCiogL+/v74+OOPMW7cOOzYsYO/I4mIiIiovWGhnYCUlBScO3cOERERiIiIQHR0NEpLS6Gvr49+/fph4MCBGDBgAHr37o2ePXvCyMhI7shEjSI3NxexsbGIiYmRPv/x8fFQKpV47LHHMHDgQGkbOnQorKys5I5M/08IARsbG6xcuRJLly6VO06jq62tRUxMDM6dO4eLFy8iPDwcV69ehVKphK2tLTw8PODh4YEBAwbAzc2NBUdqlWpqapCUlCS1weHh4YiMjERxcTH09fXRv39/eHh4YNCgQRg2bBg/563A+fPnMXfuXCgUCnz00Uf45z//KXckIiIiIqLmwkI71adUKnH16lWp8BgREYHLly+joqICGhoacHZ2hqurK9zc3ODm5gZXV1f07NkThoaGckcnuqu8vDxcuXIFcXFxiI2NRVxcHK5cuYJbt24BAMzNzaVvcbi7u2PgwIEs6LRw165dQ5cuXXDmzBkMHTpU7jjNoqSkBJGRkQgPD0d4eDguXryItLQ0AICpqSnc3NzQq1cvtc3a2lre0ET4b8dYWloaYmNjceXKFcTExODKlSu4evUqqqqqoKWlhZ49e2LQoEFSYb13795cS6aVKi8vx9q1a/Hxxx9j/Pjx2LFjBzp16iR3LCIiIiKipsZCOz0YpVKJ1NRUxMTE4OrVq1KxMj4+HlVVVdDU1ISzszNcXFzQrVs3dO3aFV27dkW3bt3QpUsXTkFDTa6wsBApKSlISUlBcnKy9Dg+Ph4KhQLAf4uRqk4iV1dX9OrVC66uriwAtEL79+/HtGnTUFhYCGNjY7njyKagoKBe8TImJga5ubkAACsrK7i6uqJbt27Spmqn+e0kamy5ublISkpCcnKy2n/j4+NRXFwMAHByclLrFFK1x/r6+jKnp8Z26tQpvPDCC8jLy8PWrVsxe/ZsuSMRERERETUlFtrp0dTU1CAlJQWxsbG4evWq9A/rlJQUabSwpqYmOnbsqFaAd3R0hKOjIzp16gR7e3vo6urKfCXU0pWVleHGjRu4efMmMjIykJqaKhXTU1JSkJOTAwDQ0tKCg4OD9FlzcXGRpj1ydHSU+SqosbzzzjvYu3cv4uPj5Y7SIikUCmlapISEBCQnJyM5ORnp6emora0FANjZ2UlF927dusHJyUlqm+3t7aGtrS3zVVBLU1FRgRs3biA9PR03b96UOjZVW35+PgBAV1dX6nx3cXFBjx49pKI6F41uX0pLS7Fy5Ups27YNU6dOxZdffgkLCwu5YxERERERNQUW2qnplJSUqBVCVf8gT01Nxc2bN1FdXQ0A0NDQgK2tLRwcHNCxY0c4ODioPba2toa9vX27HrXa1uXk5CAnJwcKhQI3b96UCuqqxxkZGcjLy5Neb2BggM6dO0vfmlAV1bt27YrOnTuz46Yd8PHxgYGBAfbs2SN3lFalsrISqampUqdo3a1uu6ylpQVbW1s4OTnBwcEBnTp1UivC29vbw8rKit9WakNKS0uRlZWFW7duIT09XSqmX79+XfpZ1aEJ/K8drttZo9ocHR2hpaUl49VQS3PixAk8//zzUCqVCAgIwIQJE+SORERERETU2FhoJ3nU1tZCoVAgPT0dGRkZSE9PlwqqqscKhQI1NTXSewwMDGBtbQ07OztYWVmpPbaxsYGdnR06dOgACwsLWFhYcM54GRUVFSE/Px/5+fnIycnBrVu3kJ2dDYVCgVu3biEnJwdZWVnIzs5GTk6OVNwDAH19fbWOFkdHR7UOmE6dOsHS0lLGq6OWwMnJCf/617+watUquaO0Gap2WVVYVRVZVR1f6enp0jeVVDp06AAbGxupQ9Ta2lpqj62trWFtbQ1LS0tYWFjA3Nxcpitrn5RKJfLz85GXl4e8vDypDVYoFMjOzkZmZiays7Nx69YtZGVlobS0VHqvtrY27Ozs1L595uDgIHW8ODg44LHHHpPx6qg1KigowMKFC7Fnzx68/PLL+Oijj9hZR0RERERtCQvt1HIplUqpIJCVlSUVbBUKRb3H2dnZ0nQIKnp6erCwsFArvqs21T5jY2OYmJjA1NQUhoaGMDQ0lIr0hoaGMDU1lenq5SGEQEFBAUpLS1FWVobi4mIUFRWhrKwMZWVlyM/PR0lJiVS8URXTVZtqn1KpVDuurq6uVICztbWFlZUVbG1tYWNjo/ZY9TPR/eTn58PS0hKHDx/GxIkT5Y7TrlRUVCAzM1OtbVZ1pGVmZkrfTFEoFCgrK1N7r6amptT2dujQ4a6PVe2xqakpjIyMYGhoCHNzcxgbG8PIyKhdzStfWFiI0tJSlJaWoqioCMXFxdLPBQUFUoemqpCen5+P3NxcaV9hYWG9Y5qZmUmdIHZ2dmqdJKq2WLVx6iBqKoGBgVi8eDEcHR3xww8/oE+fPnJHIiIiIiJqDCy0U9tQW1uLnJwcteLv3QrBd+4vKSmRFmi7lzuL8ABgYmICbW1t6OnpSSPnVc8ZGRlBV1cXOjo69aa7MTc3h4aGxl3PY2ZmBk1NzXr7q6qq1EYa1lVdXY2SkhK1fQUFBRBCoKysDJWVlaipqZGusbCwELW1taioqEB5ebn0XFFREUpLS1FeXn7f
```

Comment:
Since I am using the "replace_spec['node_sort2'] = {"load": cached_sort}", so the graph won't go up to node_csvdata as the dependence is resolved at the "node_sort2'. It is not necessary to add replace_spec['node_csvdata']. 

that's said, replace_spec is defined before and the "node_csvdata" is already set there. So replace_spec has it at this cell, though it is not necessary.

Reply:
I see. I was thrown off by the
```
action = "load" if os.path.isfile('./.cache/node_csvdata.hdf5') else "save"
```
two lines above. If the`action` is not needed or used then you should delete it.

Reply:
done